routers/web/user/profile.go

```diff
@@ -124,7 +124,7 @@ func Profile(ctx *context.Context) {
 		}
 	}
 
-	showPrivate := ctx.IsSigned && (ctx.Doer.IsAdmin || ctx.Doer.ID == ctx.ContextUser.ID)
+	showPrivate := ctx.IsSigned && (ctx.Doer.IsAdmin || ctx.Doer.ID == ctx.ContextUser.ID || !ctx.Doer.KeepActivityPrivate)
```

Comment:
User himself or admin could still visit those information.

Reply:
yes

Reply:
Are you sure the Doer is the correct user? Doer means the visitor.

Reply:
Oh sorry, corrected